scripts/base/frameworks/cluster/main.zeek

```diff
@@ -341,11 +396,25 @@ event Cluster::hello(name: string, id: string) &priority=10
 	n$id = id;
 	Cluster::log(fmt("got hello from %s (%s)", name, id));
 
+	local active_ids = active_node_ids[n$node_type];
+	add active_ids[id];
+	active_node_ids[n$node_type] = active_ids;
+
 	if ( n$node_type == WORKER )
-		{
-		add active_worker_ids[id];
-		worker_count = |active_worker_ids|;
-		}
+		worker_count = |active_node_ids[WORKER]|;
+	}
+
+event Cluster::node_up(name: string, id: string) &priority=-10
+	{
+	local mgr = nodes[node]?$manager ? nodes[node]$manager : "";
+	if ( name !in nodes_awaited && name != mgr )
```

Comment:
That guard is a bit confusing here. Is it actually needed? A `delete` for a non-existing element is a noop.

It may need a guard `|nodes_awaited| == 0` and early return to avoid triggering the events multiple times.

Reply:
There will be more `node_up`s than nodes awaited (see comment above) and `node_connected` should be triggered only once (the documentation wasn't updated, sorry). Furthermore, my thought was that a node could be connected to all its fellows before the manager is up.